src/main/java/jackson/Jackson.java

```diff
@@ -0,0 +1,225 @@
+package jackson;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+
+import jackson.enums.Actions;
+import jackson.enums.Commands;
+import jackson.exceptions.OutOfListException;
+import jackson.exceptions.SyntaxException;
+import jackson.exceptions.UnsupportedException;
+import jackson.tasks.Deadline;
+import jackson.tasks.Event;
+import jackson.tasks.Task;
+import jackson.tasks.Todo;
+import jackson.utils.Parser;
+import jackson.utils.Response;
+import jackson.utils.Storage;
+import jackson.utils.TaskList;
+import jackson.utils.Ui;
+
+/**
+ * Main class for the chatbot.
+ */
+public class Jackson {
+
+    /* Expected number of tasks to store */
+    private static final int EXPECTED_SIZE = 100;
+
+    /* Path to save list data */
+    private static final String PATH = "src/main/resources/texts/data.txt";
+
+    /* Stores secret text for greedy loading */
+    private static String secret = "";
+
+    /* Stores previous command type for css style changing */
+    private Commands.CommandType commandType;
+
+    /* Instance variables for main loop */
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Constructs Jackson instance.
+     */
+    public Jackson() {
+        this.taskList = new TaskList(EXPECTED_SIZE);
+        this.ui = new Ui();
+        this.storage = new Storage(PATH);
+        this.commandType = Commands.CommandType.INTRO;
+    }
+
+    /**
+     * Reads secret text from secret file and prints it.
+     * If secret file not found, handles exception and prints error message.
+     * @return String response.
+     */
+    public String readSecret() {
+        // get string builder to read line by line
+        StringBuilder temp = new StringBuilder();
+        String output;
+        if (secret.isEmpty()) {
+            File f = new File("src/main/resources/texts/secret_text.txt");
+            try {
+                // read via scanner line by line
+                Scanner sc = new Scanner(f);
+                while (sc.hasNextLine()) {
+                    temp.append(sc.nextLine()).append("\n");
+                }
+                output = temp.toString().strip();
+
+                // decode from base64 to utf8
+                byte[] decoded = Base64.getDecoder().decode(output);
```

Comment:
Maybe this one could add more description to the variable on what is being decoded?

Reply:
thanks for the review! however, i wanted to make this an easter egg (albeit poorly hidden) as much as possible, so had to be vague with the names here